lib/ts/ink_memory.cc

```diff
@@ -209,6 +210,46 @@ ats_mlock(caddr_t addr, size_t len)
   return res;
 }
 
+void *
+ats_track_malloc(size_t size, uint64_t *stat)
+{
+  // pad with 8 bytes to add allocation size
+  void *ptr = malloc(size + 8);
+  memcpy(ptr, &size, 8);
+  ink_atomic_increment(stat, size);
+  return (void *)((uint64_t)ptr + 8);
+}
```

Comment:
I know we only support 64 bit, but shouldn't we still use `sizeof()` and `uintptr_t`?